lib/sourcebit.js

```diff
@@ -111,44 +111,45 @@ class Sourcebit {
     return ora(message)[oraMethod]();
   }
 
-  logFromPlugin(pluginName, message, messageType) {
+  logFromPlugin(message, messageType) {
     this.log(`${message}`, messageType);
   }
 
-  parsePluginOptions(plugin) {
-    const { options } = plugin;
-
-    if (!options) return {};
+  parsePluginOptions(plugin, optionsFromConfig) {
+    const { options: optionsSchema = {} } = plugin;
 
     const defaults = {};
     const overrides = {};
 
-    Object.keys(options).forEach(key => {
-      if (options[key].default !== undefined) {
-        defaults[key] = options[key].default;
+    Object.keys(optionsSchema).forEach(key => {
+      if (optionsSchema[key].default !== undefined) {
+        defaults[key] = optionsSchema[key].default;
       }
 
       if (
-        typeof options[key].runtimeParameter === "string" &&
-        this.runtimeParameters[options[key].runtimeParameter] !== undefined
+        typeof optionsSchema[key].runtimeParameter === "string" &&
+        this.runtimeParameters[optionsSchema[key].runtimeParameter] !==
+          undefined
       ) {
-        overrides[key] = this.runtimeParameters[options[key].runtimeParameter];
+        overrides[key] = this.runtimeParameters[
+          optionsSchema[key].runtimeParameter
+        ];
       }
     });
 
-    return { defaults, overrides };
+    return Object.assign({}, defaults, optionsFromConfig, overrides);
   }
 
   saveContextToCache() {
     if (this.runtimeParameters.cache === false) return;
 
     const serializedCache = JSON.stringify(this.context);
 
-    try {
-      fs.writeFileSync(this.cacheFilePath, serializedCache);
-    } catch (error) {
-      console.log(error);
-    }
+    fs.writeFile(this.cacheFilePath, serializedCache, error => {
```

Comment:
`saveContextToCache` is called from `bootstrapAll`, not from `loadContextFromCache`. I can make this return a Promise and make `bootstrapAll` await for it, but in that case we should probably wrap it in a try/catch, because if saving to cache fails for some reason, it shouldn't throw a fatal error? That's why I opted for a "fire and forget" approach: we call it, don't wait for it and don't really do anything if it fails.

What do you think the behaviour should be?

Reply:
Sure, I can do that.